kyo-core/shared/src/main/scala/kyo/Scope.scala

```diff
@@ -0,0 +1,126 @@
+package kyo
+
+import kyo.Result.Error
+import kyo.Result.Panic
+import kyo.Tag
+import kyo.kernel.ContextEffect
+
+opaque type Scope = Sync
```

Comment:
This is planned to be the replacement for `Resource`. Instead of using `ContextEffect`, which has more strict type-level tracking, I'm using an opaque type + `Local`. The effect tracking is meant to ensure the `Local` is set in a scope to allow `runFinalizer` to dynamically check if there's an outer `Scope` finalizer.